tools/autograph-client/client.go

```diff
@@ -190,6 +193,12 @@ examples:
 			ZIP: zipMethodOption,
 		}
 	}
+	// if we have a pk7digest but no cn, assume we're signing an APK
+	if pk7digest != "" && cn == "" {
```

Comment:
nit: I think it'd be cleaner to make this an `else`

Reply:
I don't think it works, because if it's not an xpi, then it could be one on of 3 other types.

I think we should rewrite this client entirely, to be honest, and have type detection via filename or explicit flags.

Reply:
opened: https://github.com/mozilla-services/autograph/issues/163

Reply:
I think separate flags are better for now (was debating this at the time and thought a consistent CLI interface would be better but setting options shouldn't get this complicated).